packages/edit-site/src/components/app/index.js

```diff
@@ -37,10 +38,12 @@ export default function App() {
 		<SlotFillProvider>
 			<GlobalStylesProvider>
 				<UnsavedChangesWarning />
-				<RouterProvider>
-					<Layout />
-					<PluginArea onError={ onPluginAreaError } />
-				</RouterProvider>
+				<FontLibraryProvider>
+					<RouterProvider>
+						<Layout />
+						<PluginArea onError={ onPluginAreaError } />
+					</RouterProvider>
+				</FontLibraryProvider>
```

Comment:
I'm curious what is this provider about?

Reply:
Took a look at it (very quickly), it seems like a random provider to share some props and things like that. I'm not entirely sure that we conceptually need a context at all for this. Can't we work on removing it entirely?

Reply:
> Can't we work on removing it entirely?

Yes, I think most of the context here can be removed with proper refactoring.

However, I think there is one context that must be shared between the modal dialog and the typography panel. We need to dynamically switch the content of a modal dialog depending on whether I click on an icon in the typography panel or on any font. Is there a way to achieve this behavior without using a provider?

https://github.com/WordPress/gutenberg/assets/54422211/62e8f445-711c-472f-9e49-2f939d17a02b

Reply:
If we are to move the modal open/close state to the interface package, then yes, context is an option. But it feels like we could have some "initialState" argument that we can pass to `openModal`. 

In other words, how do we do this in other modals? Or maybe we didn't have a use-case in other modals to not open a modal in its initial state.

--
Regardless of this discussion, and since we have the context today, I think it's fine to add it top-level with the caveat that we should move away from it. Conceptually `edit-site/app/index` should be seen as the future wp-admin index. It seems that adding random providers there is not a great way to achieve things. "Routes" need to be lazy loaded basically.

Reply:
> Should we consider adding new arguments to openModal after removing the context as far as possible and then come back to this PR?

Yes, I'd like that ❤️